src/averageAges.js

```diff
@@ -37,7 +58,23 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let arrWomen = [];
+
+  if (withChildren === true) {
+    arrWomen = people.filter(person => person.sex === 'f')
+      .filter(woman => getChildren(people, woman).length > 0);
```

Comment:
You just need to know if a woman has any children or not.
You really don't need to know number of children.
I think, `find` method would be better for `getChildren`, so it returns just `true` or `false`.